### PR TITLE
Hygen setup

### PR DESCRIPTION
**Pull request by clarknoah**
Added Hygen Templates

### CODE REVIEW

File: browser/_templates/generator/help/index.ejs.t [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (ZakkMan) [inline browser/_templates/generator/help/index.ejs.t]: Do we need these? What purpose do they serve?

File: browser/_templates/react-boilerplate/new/index.ejs.t [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import {
+  BrowserRouter as Router
+} from "react-router-dom"
+import * as serviceWorker from './serviceWorker';
+
+ReactDOM.render(<Router><App /></Router>, document.getElementById('root'));
```
Comment (ZakkMan) [inline browser/_templates/react-boilerplate/new/index.ejs.t]: Setting up React Router should be a separate PR

File: browser/_templates/react-boilerplate/new/app.ejs.t [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (ZakkMan) [inline browser/_templates/react-boilerplate/new/app.ejs.t]: Do we need this? there's already an `App.js`

File: browser/_templates/component-const/new/reactComponent.ejs.t [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+to: src/components/<%= name %>/<%= name %>.js
+---
+
+import React from 'react';
+import "./<%= name %>.css";
+
+// Function based React Component
+const <%= name %> = (props) => {
```
Comment (ZakkMan) [inline browser/_templates/component-const/new/reactComponent.ejs.t]: Is there a way to uppercase the component names? i.e. name the component `Button` if I run `hygen component-const new button`.

File: browser/_templates/react-boilerplate/new/index.ejs.t (hunk 2 above)
Comment (ZakkMan) [reply browser/_templates/react-boilerplate/new/index.ejs.t]: Oh wait - I see what you're doing!

We're going to setup index, app, etc in a later PR - we only need a generator for repeated actions (like creating components) - good thinking though!

Comment (clarknoah) [reply browser/_templates/react-boilerplate/new/index.ejs.t]: Oh wow, this is embarrassing, the entire react-boilerplate was accidentally included from some other project I was working on, totally didn't even see that I left it in the directory!

File: browser/_templates/component-const/new/reactComponent.ejs.t (hunk 4 above)
Comment (clarknoah) [reply browser/_templates/component-const/new/reactComponent.ejs.t]: When you use Hygen (as the script is currently configured), in order to upper case the component names, you must do it when you write it into command line - EG: 
`hygen component-const new --name Button`
--> * src/Component/Button/Button.js*

I had intially set it up where it would capitalize, but if the variable name is more than one word, it will only capitalize the first character in the string,  EG: 
`hygen component-const new --name newbutton`
--> * src/Component/Newbutton/Newbutton.js*
so I decided against using it. Would you like me to change it?

**Comment by ZakkMan** [comment]
@ZakkMan

**Comment by ZakkMan** [comment]
@clarknoah
